### PR TITLE
Search: Enable search on private WPCOM simple sites

### LINKED ISSUE #20904: Jetpack Search: Network requests can arrive out-of-order for private simple WPCOM sites
### Impacted plugin

Jetpack

### Steps to Reproduce

Rapidly triggering queries, especially for a site with a large index, can result in search API responses arriving out-of-order. This only occurs when `wpcom-proxy-request` code path is used in our API code.

### Browser

Chrome/Chromium, Firefox

### PR DESCRIPTION
**Pull request by jsnmoon**
#### Changes proposed in this Pull Request:
- Removes `wpcom-proxy-request` dependency for searching private simple WPCOM sites.
- Adds support for a WPCOM origin, first-party search API route.

#### Jetpack product discussion
None.

#### Does this pull request change what data or activity we track or use?
None.

#### Testing instructions:
See D67187-code for testing instructions. These changes should have zero impact on Jetpack sites and are intended to target only simple WordPress.com sites.

### CODE REVIEW

File: projects/plugins/jetpack/modules/search/instant-search/lib/api.js [hunk 1]
```diff
@@ -406,19 +388,19 @@ export function search( options, requestId ) {
...
-	}
+	const { apiNonce, apiRoot, homeUrl, isPrivateSite, isWpcom } = window[ SERVER_OBJECT_NAME ];
 
-	// NOTE: Both atomic and Jetpack sites can be set to "private".
+	// NOTE: Both simple and atomic sites can be set to "private".
+	//       "Private" Jetpack sites are not yet supported.
 	const urlForPublicApi = `https://public-api.wordpress.com/rest/v1.3${ pathForPublicApi }`;
-	const urlForPrivateApi = `${ apiRoot }wpcom/v2/search?${ queryString }`;
-	const url = isPrivateSite ? urlForPrivateApi : urlForPublicApi;
+	const urlForWpcomOrigin = `${ homeUrl }/wp-json/wpcom-origin/v1.3${ pathForPublicApi }`;
```
Comment (kangzj) [inline projects/plugins/jetpack/modules/search/instant-search/lib/api.js:396]:
```suggestion
	const urlForWpcomOrigin = `${ apiRoot }wpcom-origin/v1.3${ pathForPublicApi }`;
```
would this work?

File: projects/plugins/jetpack/modules/search/instant-search/external/wpcom-proxy-request.js [hunk 2]
```diff
@@ -1,460 +0,0 @@
-/**
```
Comment (kangzj) [inline projects/plugins/jetpack/modules/search/instant-search/external/wpcom-proxy-request.js:1]: Yeah! a big win

**Review by kangzj — APPROVED** [review]
It works for simple, jetpack and atomic sites and looks good to me. Great job!

File: projects/plugins/jetpack/modules/search/instant-search/lib/api.js (hunk 1 above)
Comment (jsnmoon) [reply projects/plugins/jetpack/modules/search/instant-search/lib/api.js:396]: No; for simple sites, the `apiRoot` is set to `public-api.wordpress.com`, which is not desirable here.

**Review by jeherve — APPROVED** [review]

**Comment by jsnmoon** [comment]
🚢 r232294-wpcom

**Pull request merged**